edge/pkg/metamanager/metaserver/kubernetes/scope/scope.go

```diff
@@ -6,14 +6,29 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/managedfields"
 	"k8s.io/apiserver/pkg/endpoints/handlers"
 	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/kube-openapi/pkg/validation/spec"
 
 	"github.com/kubeedge/kubeedge/edge/pkg/metamanager/metaserver/kubernetes/fakers"
 	"github.com/kubeedge/kubeedge/edge/pkg/metamanager/metaserver/kubernetes/serializer"
+	"sigs.k8s.io/structured-merge-diff/v4/fieldpath"
 )
 
 func NewRequestScope() *handlers.RequestScope {
+	fakeTypeConverter, _ := managedfields.NewTypeConverter(make(map[string]*spec.Schema), false)
```

Comment:
`managedfields.NewTypeConverter` returns always success? if that returns NULL, that is gonna generate the sigsegv as following?

Reply:
`make(map[string]*spec.Schema)` is a placeholder, it is an empty map, so I think there won't be any error.
```
# kubeedge/vendor/k8s.io/apimachinery/pkg/util/managedfields/typeconverter.go
func NewTypeConverter(openapiSpec map[string]*spec.Schema, preserveUnknownFields bool) (TypeConverter, error) {
	return internal.NewTypeConverter(openapiSpec, preserveUnknownFields)
}
```
```
# kubeedge/vendor/k8s.io/apimachinery/pkg/util/managedfields/internal/typeconverter.go
func NewTypeConverter(openapiSpec map[string]*spec.Schema, preserveUnknownFields bool) (TypeConverter, error) {
	typeSchema, err := schemaconv.ToSchemaFromOpenAPI(openapiSpec, preserveUnknownFields)
	if err != nil {
		return nil, fmt.Errorf("failed to convert models to schema: %v", err)
	}

	typeParser := typed.Parser{Schema: smdschema.Schema{Types: typeSchema.Types}}
	tr := indexModels(&typeParser, openapiSpec)

	return &typeConverter{parser: tr}, nil
}
```
```
# kubeedge/vendor/k8s.io/kube-openapi/pkg/schemaconv/openapi.go
func ToSchemaFromOpenAPI(models map[string]*spec.Schema, preserveUnknownFields bool) (*schema.Schema, error) {
	c := convert{
		preserveUnknownFields: preserveUnknownFields,
		output:                &schema.Schema{},
	}

	for name, spec := range models {
		// Skip/Ignore top-level references
		if len(spec.Ref.String()) > 0 {
			continue
		}

		var a schema.Atom

		// Hard-coded schemas for now as proto_models implementation functions.
		// https://github.com/kubernetes/kube-openapi/issues/364
		if name == quantityResource {
			a = schema.Atom{
				Scalar: untypedDef.Atom.Scalar,
			}
		} else if name == rawExtensionResource {
			a = untypedDef.Atom
		} else {
			c2 := c.push(name, &a)
			c2.visitSpec(spec)
			c.pop(c2)
		}

		c.insertTypeDef(name, a)
	}

	if len(c.errorMessages) > 0 {
		return nil, errors.New(strings.Join(c.errorMessages, "\n"))
	}

	c.addCommonTypes()
	return c.output, nil
}
```

Reply:
Everything seems to be fine now. But if the NewTypeConverter function make some modifications in the future, it may return errors, which may be unnoticed at the outer layer, potentially causing issues.